cgsmiles/resolve.py

```diff
@@ -210,25 +236,49 @@ def squash_atoms(self):
             self.molecule.nodes[node_to_keep]['fragid'] += self.molecule.nodes[node_to_keep]['contraction'][node_to_remove]['fragid']
 
     def resolve(self):
+        """
+        Resolve a CGSmiles string once and return the next resolution.
+        """
+        # get the next set of fragments
+        fragment_str = next(self.fragment_strs)
```

Comment:
Oh that's why.
Since you keep track of the index (`resolution_counter`) anyway, maybe easier to just index the fragment you're looking at now?
Should this class become an iterable?

Reply:
I'm open to suggestions, but there was a logic bug so perhaps good if you took a second look at the resolve_iter